server/src/main/java/org/opensearch/gateway/remote/RemoteClusterStateService.java

```diff
@@ -367,4 +369,95 @@ private static String indexMetadataFileName(IndexMetadata indexMetadata) {
+     * @param clusterUUID uuid of cluster state to refer to in remote
+     * @param clusterName name of the cluster
+     * @return {@code Map<String, IndexMetadata>} latest IndexUUID to IndexMetadata map
+     */
+    public Map<String, IndexMetadata> getLatestIndexMetadata(String clusterUUID, String clusterName) throws IOException {
+        Map<String, IndexMetadata> remoteIndexMetadata = new HashMap<>();
+        ClusterMetadataManifest ClusterMetadataManifest = getLatestClusterMetadataManifest(clusterUUID, clusterName);
+        for (UploadedIndexMetadata uploadedIndexMetadata : ClusterMetadataManifest.getIndices()) {
+            IndexMetadata indexMetadata = getIndexMetadata(clusterUUID, clusterName, uploadedIndexMetadata);
+            remoteIndexMetadata.put(uploadedIndexMetadata.getIndexUUID(), indexMetadata);
+        }
+        return remoteIndexMetadata;
+    }
+
+    private IndexMetadata getIndexMetadata(String clusterUUID, String clusterName, UploadedIndexMetadata uploadedIndexMetadata)
+        throws IOException {
+        try {
+            return INDEX_METADATA_FORMAT.read(
+                indexMetadataContainer(clusterName, clusterUUID, uploadedIndexMetadata.getIndexUUID()),
+                uploadedIndexMetadata.getUploadedFilename(),
+                blobStoreRepository.getNamedXContentRegistry()
+            );
+        } catch (IOException e) {
+            String errorMsg = String.format(
+                Locale.ROOT,
+                "Error while downloading IndexMetadata - %s",
+                uploadedIndexMetadata.getUploadedFilename()
+            );
+            logger.error(errorMsg, e);
+            throw new IllegalStateException(errorMsg);
+        }
+    }
+
+    /**
+     * Fetch latest ClusterMetadataManifest from remote state store
+     * @param clusterUUID uuid of cluster state to refer to in remote
+     * @param clusterName name of the cluster
+     * @return ClusterMetadataManifest
+     */
+    public ClusterMetadataManifest getLatestClusterMetadataManifest(String clusterUUID, String clusterName) {
+        String latestMarkerFileName = getLatestManifestFileName(clusterUUID, clusterName);
+        return fetchRemoteClusterMetadataManifest(latestMarkerFileName, clusterUUID, clusterName);
+    }
+
+    /**
+     * Fetch latest ClusterMetadataManifest file from remote state store
+     * @param clusterUUID uuid of cluster state to refer to in remote
+     * @param clusterName name of the cluster
+     * @return latest ClusterMetadataManifest filename
+     */
+    private String getLatestManifestFileName(String clusterUUID, String clusterName) throws IllegalStateException {
+        try {
+            List<BlobMetadata> markerFilesMetadata = manifestContainer(clusterUUID, clusterName).listBlobsByPrefixInSortedOrder(
+                "manifest" + DELIMITER,
+                1,
+                BlobContainer.BlobNameSortOrder.LEXICOGRAPHIC
+            );
```

Comment:
please add a comment explaining why the top one would be latest based on the file naming.